tensorflow/java/src/main/java/org/tensorflow/op/annotation/Operator.java

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation used by classes that should be added to the Ops convenience API.
+ *
+ * <p>An annotation processor builds the {@code Ops} API by aggregating all classes annotated as
+ * {@code @Operator}s. Each annotated class <b>must</b> have at least one public static factory
+ * method named {@code create} that accepts a {@link org.tensorflow.op.Scope} as its first argument.
+ * The processor then adds a convenience method in the {@code Ops} API. For example:
+ *
+ * <pre>{@code
+ * @Operator
+ * public final class MyOp implements Op {
+ *   public static MyOp create(Scope scope, Operand operand) {
+ *     ...
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>results in a method in the {@code Ops} API
+ *
+ * <pre>{@code
+ * ops.myOp(operand);
+ * // and has exactly the same effect as calling
+ * // MyOp.create(ops.getScope(), operand);
+ * }</pre>
+ */
+@Documented
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.CLASS)
+public @interface Operator {
+  /**
+   * Specify an optional group within the {@code Ops} API
+   *
+   * <p>By default, an annotation processor will create convenience methods directly in the {@code
+   * Ops} class. An annotated operator may optionally choose to place the method within a group. For
+   * example:
+   *
+   * <pre>{@code
+   * @Operator(group="math")
+   * public final class Add extends PrimitiveOp implements Operand {
+   *   ...
+   * }
+   * }</pre>
+   *
+   * <p>results in the {@code add} method placed within a {@code math} group within the {@code Ops}
+   * API.
+   *
+   * <pre>{@code
+   * ops.math().add(...);
+   * }</pre>
+   */
+  String group() default "";
```

Comment:
Yup, only adding it later if and when we need it sounds good. Was just raising the question to think about (and also perhaps suggest that the `AnnotationProcessor` would want to ensure that there are no `.`s in the `String` returned by `group()` etc.)